### PR TITLE
Unify arch-specific Dockerfiles into multi-arch one

### PR DESCRIPTION
**Pull request by arm64b**
Now we have several Dockerfiles for all the available platforms, the contents in those files are almost the same, naturally one solution is to consolidate all those Dockerfiles into only one, which will ease the effort of the code maintenance, for example, we have multi files need to be updated if we want to bump the version of `runc`.    

**- What I did**
Remove the redundancy of the Dockerfile 
**- How I did it**
Consolidate the different arch-specific Dockerfiles into only one.
**- How to verify it**
`make binary` and `make test-integration`
**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by arm64b** [comment]
@vdemeester @thaJeztah @cpuguy83 @tonistiigi

**Comment by thaJeztah** [comment]
oooooh :heart_eyes: nice!

ping @tianon @seemethere @andrewhsu as well :-)

**Comment by vdemeester** [comment]
Updating the powerpc and z to "handle" that PR to make sure it works :angel: (without making other PRs fail)

```bash
# powerpc
test -f Dockerfile.ppc64le && \
docker build --rm --force-rm --build-arg APT_MIRROR=cdn-fastly.deb.debian.org -t docker-powerpc:$GITCOMMIT -f Dockerfile.ppc64le . || \
docker build --rm --force-rm --build-arg APT_MIRROR=cdn-fastly.deb.debian.org -t docker-powerpc:$GITCOMMIT -f Dockerfile .
# z
test -f Dockerfile.s390x && \
docker build --rm --force-rm --build-arg APT_MIRROR=cdn-fastly.deb.debian.org -t docker-s390x:$GITCOMMIT -f Dockerfile.s390x . || \
docker build --rm --force-rm --build-arg APT_MIRROR=cdn-fastly.deb.debian.org -t docker-s390x:$GITCOMMIT -f Dockerfile .
```

**Comment by arm64b** [comment]
Thanks @vdemeester and @thaJeztah for the quick response, especially the command line for `powerpc` and `z` nodes is extremely helpful and magic 👼

**Review by dnephin — APPROVED** [review]
LGTM

**Review by kolyshkin — CHANGES_REQUESTED** [review]
CRIU for other platforms, also s/Malti-arch/Multiarch/

**Comment by seemethere** [comment]
Going to try and see if this will work with our build/release pipeline for Docker CE.

**Review by seemethere — APPROVED** [review]
LGTM

File: Dockerfile [hunk 1]
```diff
@@ -57,7 +59,12 @@ RUN apt-get update && apt-get install -y \
 	&& curl -sSL https://github.com/checkpoint-restore/criu/archive/v${CRIU_VERSION}.tar.gz | tar -C /usr/src/criu/ -xz --strip-components=1 \
 	&& cd /usr/src/criu \
 	&& make \
-	&& make PREFIX=/opt/criu install-criu
+	&& make PREFIX=/opt/criu install-criu ;\
+	;; \
+    armv7l|aarch64|ppc64le|s390x) \
+	mkdir -p /opt/criu; \
```
[the remark this answers is not in the document]
Comment (arm64b) [reply Dockerfile]: The typo has been fixed. For the CRIU stuff, thanks for the suggestion 👍 , so I'll submit a separate PR to enable CRIU on ARM, PPC64 and S390x platforms. This PR just focuses on the build pipeline.

**Review by thaJeztah — APPROVED** [review]
LGTM

> I'll submit a separate PR to enable CRIU on ARM, PPC64 and S390x platforms after this PR merged. This PR just focuses on the build pipeline.

Agreed, let's keep that change separate from this one (thanks @kolyshkin for providing the info 👍)

**Review by vdemeester — APPROVED** [review]
LGTM :tiger:

**Pull request merged**